app/controllers/teams_controller.rb

```diff
@@ -59,57 +55,76 @@ def destroy
 
   # Assigns team ownership to a specific team member
   def assign_ownership
-    unless owners_include_user
-      team_owners << user
-      redirect_to @team
-    end
+    assign_ownership_to_member
+    redirect_to @team
   end
 
   def delete_ownership
-    delete_owner_if_existing
+    delete_ownership_from_member
     redirect_to @team
   end
 
   def delete_membership
-    delete_owner_if_existing
-    @team.members.delete(user)
+    delete_membership_from_member
     redirect_to @team
   end
 
-  def current_ability
-    @current_ability ||= Ability.new(current_user, params[:team_member])
+  def perform_action_on_multiple_members
+    # choose action based on submit button that was pressed
+    if params[:assign_ownership]
+      @action = "assign_ownership_to_member"
+    elsif params[:delete_ownership]
+      @action = "delete_ownership_from_member"
+    elsif params[:delete_membership]
+      @action = "delete_membership_from_member"  
+    end
+
+    params[:members].each do |member|
+      params[:team_member] = member          
+      begin
+        send(@action)   
+      rescue => ex
+        print ex.message
+      end
+    end
+    redirect_to @team
   end
 
   private
-    def delete_owner_if_existing
-      if owners_include_user
-        team_owners.delete user
-      end
+    def assign_ownership_to_member
+      authorize! :assign_ownership, @team      
+      @member_to_become_owner = TeamUser.find_by(user_id: user, team_id: @team.id)
+      @member_to_become_owner.update_attribute(:is_owner, true)
+    end
+
+    def delete_ownership_from_member
+      authorize! :delete_ownership, @team  
+      @member_to_become_owner = TeamUser.find_by(user_id: user, team_id: @team.id)
```

Comment:
Please find a better name for this variable – seems like copied from the other method

Reply:
^^